lib/capybara/session.rb

```diff
@@ -85,7 +85,7 @@ def initialize(mode, app = nil)
       end
       @server = if config.run_server && @app && driver.needs_server?
         server_options = { port: config.server_port, host: config.server_host, reportable_errors: config.server_errors }
-        server_options[:extra_middleware] = [Capybara::Server::AnimationDisabler] if config.disable_animation
+        server_options[:extra_middleware] = [Capybara::Server::AnimationDisabler]
```

Comment:
Why are you always loading AnimationDisabler middleware?

Reply:
one approach: load the middleware conditionally, and have the animation setting be per-session.

another approach: load the middleware always, but have it do something (or not) based on the configuration value.

Reply:
This then breaks per session configuration ( "threadsafe" mode )